### ISSUE TITLE
Avoid replacing so many characters with `_` in `_parse_raw_sam_df()`

### ISSUE THREAD

**Issue by Peque** [pvlib/pvlib-python#768]
As [suggested by @wholmgren](https://github.com/pvlib/pvlib-python/pull/753#issuecomment-513292390).

I too think it would be more convenient. For example, it would allow for easier distinction between manufacturer and module number.

**Comment by Peque** [comment]
@wholmgren @cwhanse Since we already broke the API by changing the module and inverter databases (i.e.: modules/inverters name need to be updated to work with 0.7), would you be okay to introduce this change as well?

Asking before spending time preparing the PR. :blush:

**Comment by Peque** [comment]
Just adding [another vote](https://github.com/pvlib/pvlib-python/pull/753#issuecomment-513456822) in favor of this change:

> @adriesse 
> 
> I've never use the tab completion and find the database more convenient in its non-transposed form.

**Comment by cwhanse** [comment]
I agree with making changes to the module dataframe keys. I think our goal is to remain as consistent as possible with the [source](https://github.com/NREL/SAM/blob/develop/deploy/libraries/CEC%20Modules.csv). We can propose changes to the source, if that's the best course. The SAM developers agree with the overall goal to have one name for a module that is common among both codes.

**Comment by Peque** [comment]
@cwhanse @wholmgren I would propose to have a separator between the manufacturer name and the model name, either a `,` (separate columns in the CSV) or some other character. It used to be the case (a `:` was used), but it seems not anymore. :confused: 

I opened this issue, in case you want to upvote/comment: https://github.com/NREL/SAM/issues/227

I may wait for that to be resolved before implementing this change (feeling lazy, in case I have to change the names 2 times... :joy:)

**Comment by cwhanse** [comment]
@Peque looking for closure on v0.7 release. Would you prefer to defer work in pvlib on this issue, pending any changes in SAM?

**Comment by Peque** [comment]
@cwhanse Thanks for the ping.

I would prefer to defer work pending changes in SAM, if you do not mind.

**Comment by cwhanse** [comment]
@Peque don't mind at all.

**Comment by Peque** [comment]
Maybe related (creating a separate quick-to-load Python package for parts): https://groups.google.com/forum/#!topic/pvlib-python/Sqe84M2C6RY
